src/main/java/uk/gov/pay/adminusers/service/InviteService.java

```diff
@@ -62,14 +62,15 @@ public void reGenerateOtp(InviteOtpRequest inviteOtpRequest) {
             inviteDao.merge(invite);
             int newPassCode = secondFactorAuthenticator.newPassCode(invite.getOtpKey());
             String passcode = String.format(Locale.ENGLISH, SIX_DIGITS_WITH_LEADING_ZEROS, newPassCode);
-            notificationService.sendSecondFactorPasscodeSms(inviteOtpRequest.getTelephoneNumber(), passcode)
-                    .thenAcceptAsync(notificationId -> LOGGER.info("sent 2FA token successfully for invite code [{}], notification id [{}]",
-                            inviteOtpRequest.getCode(), notificationId))
-                    .exceptionally(exception -> {
-                        LOGGER.error(format("error sending 2FA token for invite code [%s]", inviteOtpRequest.getCode()), exception);
-                        return null;
-                    });
+            
             LOGGER.info("New 2FA token generated for invite code [{}]", inviteOtpRequest.getCode());
+            
+            try {
+                String notificationId = notificationService.sendSecondFactorPasscodeSms(inviteOtpRequest.getTelephoneNumber(), passcode);
+                LOGGER.info("sent 2FA token successfully for invite code [{}], notification id [{}]", inviteOtpRequest.getCode(), notificationId);
+            } catch (Exception e) {
+                LOGGER.error(format("error sending 2FA token for invite code [%s]", inviteOtpRequest.getCode()), e);
+            }
```

Comment:
Presumably this was reordered to log before sending the email because the previous async approach means the log line would be written before the email sending had finished anyway?

Reply:
That's right.